tests/MetadataTest.php

```diff
@@ -56,20 +58,90 @@ public function testMetadataFilesReturnArrays()
         }
     }
 
-
     public function testIDPRemoteMetadataIDPCode()
     {
         $idpEntries = Metadata::getIdpMetadataEntries($this->metadataPath);
-        $idpCode = 'IDPNamespace';
 
         foreach($idpEntries as $entityId => $entry) {
-            $this->assertTrue(isset($entry[$idpCode]), 'Metadata entry does not include an ' . $idpCode .
-                                                  ' element as expected. IDP: ' . $entityId);
-            $this->assertTrue(is_string($entry[$idpCode]), 'Metadata entry has an IDPCode element that is not ' .
-                                                     'a string. IDP: ' . $entityId);
-            $this->assertRegExp("/^[A-Za-z0-9_-]+$/", $entry[$idpCode], 'Metadata entry has an ' .
-                                $idpCode .' element that has something other than letters, ' .
-                                'numbers, hyphens and underscores. IDP: ' . $entityId);
+            $this->assertTrue(isset($entry[self::IdpCode]), 'Metadata entry does not ' .
+                'include an ' . self::IdpCode . ' element as expected. IDP: ' . $entityId);
+
+            $nextCode = $entry[self::IdpCode];
+            $this->assertTrue(is_string($nextCode), 'Metadata entry has an ' . self::IdpCode .
+                self::IdpCode . 'element that is not a string. IDP: ' . $entityId);
+            $this->assertRegExp("/^[A-Za-z0-9_-]+$/", $nextCode, 'Metadata entry has an ' .
+                self::IdpCode .' element that has something other than letters, ' .
+                'numbers, hyphens and underscores. IDP: ' . $entityId);
+        }
+    }
+
+    public function testIDPRemoteMetadataBadSPList()
+    {
+        $hubMode = Env::get('HUB_MODE', true);
+        if ( ! $hubMode) {
+            $this->markTestSkipped('Skipping test because HUB_MODE = false');
```

Comment:
So are the tests being run twice, once with HUB_MODE set to true and once with it set to false?